ext/native/util/random/rng.h

```diff
@@ -15,7 +15,7 @@ class GMRng {
 		m_z = ~seed;
 		if (!m_z) m_z = 31337;
 	}
-	uint32_t R32() {
```

Comment:
This intentionally clips the value to 32 bits.

A way that might avoid the warning might be:

    return (uint32_t)((m_z << 16) + m_w); 

perhaps. Or 

    return ((m_z << 16) + m_w) & 0xFFFFFFFF

Either way, it's hardly important and I would prefer to just revert the changes made to this file.